app/celery/letters_pdf_tasks.py

```diff
@@ -1,424 +0,0 @@
-
-        if e.response is not None and e.response.status_code == 400:
-            message = f'sanitise_precompiled_pdf validation error for notification: {notification.id}.'
-            if 'message' in response.json():
-                message += ' ' + response.json()['message']
-
-            current_app.logger.error(message)
-            return None
-
-        try:
-            current_app.logger.error('sanitise_precompiled_pdf failed for notification: %s', notification.id)
-            self.retry(queue=QueueNames.RETRY)
-        except MaxRetriesExceededError:
-            current_app.logger.error(
-                'RETRY FAILED: sanitise_precompiled_pdf failed for notification %s', notification.id
-            )
-            notification.status = NOTIFICATION_TECHNICAL_FAILURE
-            dao_update_notification(notification)
-            raise
-
-
-@notify_celery.task(name='process-virus-scan-failed')
-def process_virus_scan_failed(filename):
-    move_failed_pdf(filename, ScanErrorType.FAILURE)
-    reference = get_reference_from_filename(filename)
-    notification = dao_get_notification_by_reference(reference)
-    updated_count = update_letter_pdf_status(reference, NOTIFICATION_VIRUS_SCAN_FAILED, billable_units=0)
-
-    if updated_count != 1:
-        raise Exception(
-            'There should only be one letter notification for each reference. Found {} notifications'.format(
-                updated_count
-            )
-        )
-
-    error = VirusScanError('notification id {} Virus scan failed: {}'.format(notification.id, filename))
-    current_app.logger.exception(error)
-    raise error
-
-
-@notify_celery.task(name='process-virus-scan-error')
-def process_virus_scan_error(filename):
-    move_failed_pdf(filename, ScanErrorType.ERROR)
-    reference = get_reference_from_filename(filename)
-    notification = dao_get_notification_by_reference(reference)
-    updated_count = update_letter_pdf_status(reference, NOTIFICATION_TECHNICAL_FAILURE, billable_units=0)
-
-    if updated_count != 1:
-        raise Exception(
-            'There should only be one letter notification for each reference. Found {} notifications'.format(
-                updated_count
-            )
-        )
-    error = VirusScanError('notification id {} Virus scan error: {}'.format(notification.id, filename))
-    current_app.logger.exception(error)
-    raise error
-
-
-def update_letter_pdf_status(
-    reference,
-    status,
-    billable_units,
-):
-    return dao_update_notifications_by_reference(
-        references=[reference],
-        update_dict={'status': status, 'billable_units': billable_units, 'updated_at': datetime.utcnow()},
-    )[0]
-
-
-def replay_letters_in_error(filename=None):
-    # This method can be used to replay letters that end up in the ERROR directory.
-    # We had an incident where clamAV was not processing the virus scan.
-    if filename:
-        move_error_pdf_to_scan_bucket(filename)
```

Comment:
`move_error_pdf_to_scan_bucket` can be removed from `app.letters.utils.py` as it's only called there.